### DISCUSSION TITLE
How to get the current and potential values for each measurement?

### DISCUSSION THREAD

**Discussion by jcmefra** [gimli-org/pyGIMLi#669] [Q&A]
Hello! I'm using pygimli to generate synthetic data, how do I get the current and the potential for each electrode for each measurement? I need the potential field, the surface potential and also the current and potential for each measurement, but I'm only getting the rhoa in the simulated dataset.

I did the following to get the potential:

```
scheme = ert.createData(elecs=np.linspace(start=-15, stop=15, num=4),
                                         schemeName='slm') 

field = ert.simulate(mesh, scheme=scheme, res=rhomap, 
                    sr=False, returnFields=True, verbose=True)

pot = field[2]-field[1]

ax, _ = pg.show(mesh, data=pot, cMap="RdBu_r",
                orientation='horizontal', label='Potential $u$',
                nCols=16, nLevs=9, showMesh=True, colorBar=False)

cbar = plt.colorbar(ax.collections[0], ax=ax, orientation='vertical')

#drawStreams(ax, mesh, pot, color='Black')

plt.show() #Show potential surfaces

```
![image](https://github.com/gimli-org/gimli/assets/64992303/00f638a6-0240-43d2-9bbf-73bacfd56596)

I was able to get the surface potential vs X as well (I assume this is the numerical solution using Fourier):

```
surface_nodes = np.array([node for node in mesh.nodes() if node.y() == 0.0])
surface_potentials = np.array([pot[node.id()] for node in surface_nodes])
surface_x = np.array([node.x() for node in surface_nodes])

plt.figure(figsize=(10, 6))
plt.scatter(surface_x, np.abs(surface_potentials))
plt.xlabel('x-position (m)')
plt.ylabel('Surface Potential (V)')
plt.grid(True)
plt.show()
```

![image](https://github.com/gimli-org/gimli/assets/64992303/ea6d7b7c-ac13-409a-a3cf-46e79530c587)

I suppose that with those potential values for each electrode array, the rhoa values and the geometric factor I would be able to get the current, but since I need over 500 randomized synthetic datasets, I'd like to know if I can just get the simulated data with current (I) and potential (u) values as well. Thank you in advance!

**Comment by halbmy** [comment]
> how do I get the current and the potential for each electrode for each measurement

I am not sure what you mean with the current I. We simulate the potentials for each electrode with a unit current I=1A. Or do you mean the current density distribution as shown in https://www.pygimli.org/_examples_auto/3_ert/plot_04_ert_2_5d_potential.html ?

**Reply by jcmefra** [reply]
Hello, thank you for your answer, yes that's what I was looking for (and also the confirmation of I=1A). Also I want to know if I can add the potential values for each measurement to the .dat simulated file or do I need to calculate it manually. Thank you in advance.

**Reply by halbmy** [reply]
I guess by default, only the resistance field (or the apparent resistivtity, if geometric factor is given) is filled, but it is identical to the voltage as the current is 1A.
